contracts/meta-aggregator/MetaAggregatorSwapContract.sol

```diff
@@ -120,23 +150,68 @@ contract MetaAggregatorSwapContract is
             receiver,
             isDelegate
         );
+        emit ETHSwappedForToken(amountOut, address(tokenOut), receiver);
+    }
+
+    /**
+     * @dev Swaps ETH using delegatecall.
+     * This function is called when the swap needs to happen in a delegatecall context.
+     * @param tokenIn must be the native token.
+     * @param tokenOut The ERC20 token to swap to.
+     * @param aggregator The address of the aggregator to use for the swap.
+     * @param swapData The data required for the swap.
+     * @param amountIn The amount of ETH to swap.
+     * @param minAmountOut The minimum amount of tokenOut expected.
+     * @param receiver The address to receive the tokenOut.
+     * @param isDelegate Indicates if the swap is in a delegatecall context.
+     * @return amountOut The amount of tokenOut received.
+     */
+    function swapETHDelegate(
+        address tokenIn,
+        IERC20 tokenOut,
+        address aggregator,
+        bytes calldata swapData,
+        uint256 amountIn,
+        uint256 minAmountOut,
+        address receiver,
+        bool isDelegate
+    ) external payable returns (uint256 amountOut) {
```

Comment:
Add `nonReentrant`

Reply:
This one is tricky would the calling contract have nonReentrant in its contract. Would the storage place for the nonReentrant status variable aline with our contract storage. That is why I didn't use nonReentrant modifier here.